### PR TITLE
Add versions 3.2.0 3.1.0 for protobuf.

### PR DESCRIPTION
**Pull request by weijianwen**

### CODE REVIEW

File: var/spack/repos/builtin/packages/protobuf/package.py [hunk 1]
```diff
@@ -31,6 +31,7 @@ class Protobuf(AutotoolsPackage):
     homepage = "https://developers.google.com/protocol-buffers"
     url      = "https://github.com/google/protobuf/releases/download/v2.5.0/protobuf-2.5.0.tar.bz2"
 
+    version('3.1.0', '14a532a7538551d5def317bfca41dace', url='https://github.com/google/protobuf/archive/v3.1.0.tar.gz')
```
Comment (davydden) [inline var/spack/repos/builtin/packages/protobuf/package.py]: it looks like we can create `url` for version and use `https://github.com/google/protobuf/archive/vX.Y.Z.tar.gz` for anything newer than `2.5.0` and `https://github.com/google/protobuf/releases/download/vX.Y.Z/protobuf-X.Y.Z.tar.bz2` for `2.5.0`. Do you want to give it a try?

Comment (weijianwen) [inline var/spack/repos/builtin/packages/protobuf/package.py]: Yes, it works.

**Comment by davydden** [comment]
@weijianwen can you fetch `2.5.0` with the current change? if so, looks good to me.

**Comment by weijianwen** [comment]
Yes. `protobuf@2.5.0` can be retrieved and built. Some more modifications:

1. Correct the md5sum for `protobuf@2.5.0` after url is updated
2. Add m4, autoconf, automake, libtool for all versions of protobuf since the `configure` file needs to be regenerated for the tarball from github.

**Comment by weijianwen** [comment]
Wait a moment. I am fixing some checksum errors.

**Comment by weijianwen** [comment]
It should be fine now.

**Pull request merged**